reconcile/openshift_saas_deploy_trigger_base.py

```diff
@@ -63,12 +67,158 @@ def setup(options):
     return saasherder, jenkins_map, oc_map, settings
 
 
-def construct_tekton_trigger_resource(saas_file_name,
-                                      env_name,
-                                      timeout,
-                                      settings,
-                                      integration,
-                                      integration_version):
+def trigger(spec,
+            dry_run,
+            jenkins_map,
+            oc_map,
+            already_triggered,
+            settings,
+            state_update_method,
+            integration,
+            integration_version):
+    """Trigger a deployment according to the specified pipelines provider
+
+    Args:
+        spec (dict): A trigger spec as created by saasherder
```

Comment:
Open question for the future: how do you feel about type hints? At least with them we could get a static analyzer to tell us when the callers don't obey this specification.